core/driver/sources/src/main/scala/org/virtuslab/ideprobe/ide/intellij/IntelliJPaths.scala

```diff
@@ -39,3 +41,31 @@ final class IntelliJPaths(root: Path, headless: Boolean) {
       .makeExecutable()
   }
 }
+
+object IntelliJPaths {
+  def apply(root: Path, headless: Boolean): IntelliJPaths = {
+    val bin = root.resolve("bin")
+    val ideaProperties = Config.fromFile(bin.resolve("idea.properties"))
```

Comment:
Don't we read those paths from `idea.properties` only to write them again to this file? How about letting the user specify those via config? The other problem that I can spot is that there can be some useful properties other than those paths in `idea.properties` that are going to be wiped out.

Reply:
Implementation around ij paths and properties is not very straightforward now. 
@MichalBednarz for DownloadedIJ, we don't need to care about location of specific directories like plugins or logs, it could be configurable, currently we can only configure where IJ will be installed along with all of these paths. I'd say it is good enough. As for LocalIJ, location of these paths is already configured in idea.properties, we should just read it and assume that e.g. plugins are wherever this file specifies. In such case we can just read paths from file and use it, no need for further rewrites. If this works for our windows use case, I'd keep it like that for now. However there is more to it, as idea.properties is not the only place that can be used to specify these directories I think. `-Didea.plugin.paths=something-else` would overwrite this, it could be added on invocation or in vmoptions file that can be located in couple of places with different priorities for reading it... But if we can trust the default way for our use cases, let's just do this.

Now to the refactoring:
Let's be consistent now and now make IJPaths a data class and put all setup logic inside companion object factory methods. 
We need 2 ways of creating it:
- read from config file the locations of 4 paths based on `root` and just reference all other paths to hard coded values, like the bin and executable paths.(for local)
- use predefined directories and make sure they are created (for downloaded)

We have additional responsibility here of setting up executable file content and privacy policy. It was here in more accidental way I'd say. It is awkward that IJPaths constructor takes `headless` parameter, it has nothing to do with paths. It would make sense to move the logic that sets up policy and executable to the code that is installing intellij (I know the original bad design was not yours). Replacing executable should happen for both Downloaded and Local, not sure about privacy policy, but I would set it up as well.

Reply:
`IntelliJPaths` were made a case class, method for reading them from file was implemented, `headless` parameter was moved to `InstalledIntelliJ`. Right now the `LocalIntelliJ` reads the config from `idea.properties`, filling in the gaps with defaults if something is not found, then appends all the settings to the file (in this way we don't overwrite anything we don't recognize). At cleanup the file is restored from backup.